src/vs/workbench/contrib/chat/browser/agentSessions/agentSessionsActions.ts

```diff
@@ -491,9 +491,17 @@ export class DeleteAllLocalSessionsAction extends Action2 {
 		const chatService = accessor.get(IChatService);
 		const widgetService = accessor.get(IChatWidgetService);
 		const dialogService = accessor.get(IDialogService);
+		const agentSessionsService = accessor.get(IAgentSessionsService);
+
+		const localSessionsCount = agentSessionsService.model.sessions.filter(session => isLocalAgentSessionItem(session)).length;
+		if (localSessionsCount === 0) {
+			return;
+		}
 
 		const confirmed = await dialogService.confirm({
-			message: localize('deleteAllChats.confirm', "Are you sure you want to delete all local workspace chat sessions?"),
+			message: localSessionsCount === 1
+				? localize('deleteAllChats.confirmSingle', "Are you sure you want to delete 1 local workspace chat session?")
+				: localize('deleteAllChats.confirm', "Are you sure you want to delete {0} local workspace chat sessions?", localSessionsCount),
```

Comment:
Curious why specialize between 1 and n ? The message seems to be same except for the numbers..